src/Server/GRPCServer.cpp

```diff
@@ -320,8 +320,14 @@ namespace
 
         Poco::Net::SocketAddress getClientAddress() const
         {
-            String peer = grpc_context.peer();
-            return Poco::Net::SocketAddress{peer.substr(peer.find(':') + 1)};
+            /// Returns a string like ipv4:127.0.0.1:55930 or ipv6:%5B::1%5D:55930
```

Comment:
Can't we check that and throw an exception if that's not true before using `uri_encoded_peer.find(':')`?

Reply:
Check what exactly? I guess we can check for `(ipv4|ipv6):` prefix.

Reply:
Yes, I meant checking that the string actually has that prefix.